api/handler.go

```diff
@@ -142,6 +147,84 @@ func getContainerInfoRequest(body io.ReadCloser) (*info.ContainerInfoRequest, er
 	return &query, nil
 }
 
+// request is the rawQuery portion of the url input by the user. It should have
+// length 1, otherwise the url was ill formatted (backslashes must have
+// been inserted in the argument section of the url) and function will return an
+// error. The user can set any or none of the following arguments in any order
+// with any twice defined arguments being assigned the first value.
+// If the value type for the argument is wrong the field will be assumed to be
+// unassigned
+// bools: all_time, subcontainers, oom_events, creation_events, deletion_events
+// ints: max_events, start_time (unix timestamp), end_time (unix timestamp)
+// example request: ["oom_events=true&max_events=10"]
+func getEventRequest(request []string) (*events.Request, bool, error) {
+	if len(request) <= 0 || len(request[0]) == 0 {
+		return events.NewAllEventsRequest(), true, nil
+	} else if len(request) > 1 {
+		return nil, false, errors.New(
+			"should have received one url rawQuery but got multiple")
+	}
+	rawQuery := strings.Trim(request[0], "/ ")
+
+	query := events.NewRequest()
+	getEventsFromAllTime := false
+
+	urlMap, err := url.ParseQuery(rawQuery)
+	if err != nil {
+		glog.Errorf("url parse error of %v is %v", rawQuery, err)
+		return nil, false, err
+	}
+
+	for key, val := range urlMap {
+		if key == "all_time" {
```

Comment:
What about we check each of these instead of checking in a loop:

```Go
if val, ok := urlMap["all_time"]; ok {
  getEventsFromAllTime = strconv.ParseBool(val[0])
}
```